apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,672 @@
+-module(fyler_category_manager).
+-behaviour(gen_fsm).
+
+-include("../include/log.hrl").
+-include("fyler.hrl").
+
+-define(Regex, "\"PrivateIpAddress\": \"(?<ip>[^\"]*)\"").
+-define(Node(Category, Ip), list_to_atom(lists:flatten(io_lib:format("fyler_~p_pool@~s", [Category, Ip])))).
+-define(Timeout, 60000).
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+
+-export([
+  start_link/1,
+  run_task/2,
+  cancel_task/2,
+  task_accepted/3,
+  task_rejected/3,
+  pool_connected/2,
+  pool_down/2,
+  pool_enabled/2,
+  pool_disabled/2
+]).
+
+%% ------------------------------------------------------------------
+%% gen_fsm Function Exports
+%% ------------------------------------------------------------------
+
+-export([init/1, handle_event/3,
+         handle_sync_event/4, handle_info/3,
+         terminate/3, code_change/4]).
+
+-export([start/2, zero/2, tasks/2, pools/2]).
+
+-record(state, {
+  category :: atom(),
+  retasks = fyler_queue:new() :: fyler_queue:fyler_queue(task()),
+  tasks = fyler_queue:new() :: fyler_queue:fyler_queue(task()),
+  task_filter = [] :: [task()],
+  pools = [] :: [{atom(), enabled | disabled | {pending, reference(), non_neg_integer()}}],
+  passive_pools = [] :: [atom()],
+  index = 0 :: non_neg_integer(),
+  priorities = #{} :: #{atom() => pos_integer()},
+  node_to_id = #{} :: #{atom() => iolist()},
+  check_ref :: reference()
+}).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(Category) ->
+  gen_fsm:start_link(?MODULE, [Category], []).
+
+run_task(Manager, #task{} = Task) ->
+  gen_fsm:send_event(Manager, {run, Task}).
+
+cancel_task(Manager, Id) ->
+  gen_fsm:send_event(Manager, {cancel, Id}).
+
+task_accepted(Manager, Ref, Node) ->
+  gen_fsm:send_event(Manager, {task_accepted, Ref, Node}).
+
+task_rejected(Manager, Ref, Node) ->
+  gen_fsm:send_event(Manager, {task_rejected, Ref, Node}).
+
+pool_connected(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_connected, Node}).
+
+pool_down(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_down, Node}).
+
+pool_enabled(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_enabled, Node}).
+
+pool_disabled(Manager, Node) ->
+  gen_fsm:send_event(Manager, {pool_disabled, Node}).
+
+%% ------------------------------------------------------------------
+%% gen_fsm Function Definitions
+%% ------------------------------------------------------------------
+
+init([Category]) ->
+  self() ! start,
+  {ok, start, #state{category = Category}}.
+
+start(Event, #state{} = State) ->
+  {next_state, NextState, NewState} = start_(Event, State),
+  erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}]).
```

Comment:
Line 91 is too long:   erlang:apply(?MODULE, NextState, [start, NewState#state{check_ref = undefined}])..